projects/ngx-sub-form/src/lib/ngx-sub-form.component.ts

```diff
@@ -155,13 +175,55 @@ export abstract class NgxSubFormComponent<ControlInterface, FormInterface = Cont
       return;
     }
 
-    this.formGroup.setValue(this.transformToFormGroup(obj), {
+    const transformedValue: FormInterface = this.transformToFormGroup(obj);
+
+    // for now we throw an error if the transformed value isn't an object with all the expect values
+    // there's an issue to track support for an array https://github.com/cloudnc/ngx-sub-form/issues/9
+    this.throwIfArray(transformedValue);
+    this.throwIfMissingKey(transformedValue);
+
+    this.formGroup.setValue(transformedValue, {
       emitEvent: false,
     });
     this.formGroup.markAsPristine();
     this.formGroup.markAsUntouched();
   }
 
+  private throwIfArray(transformedValue: any): void {
+    if (Array.isArray(transformedValue)) {
+      throw new ArrayNotTransformedBeforeWriteValueError();
+    }
+  }
+
+  private throwIfMissingKey(transformedValue: FormInterface) {
+    const controlKeys = (Object.keys(this.formControlNames) as any) as (keyof FormInterface)[];
+
+    // @todo why controlKeys is an empty array here?!
+    // const missingKeys: (keyof FormInterface)[] = this.controlKeys.reduce(
```

Comment:
You can try to uncomment that block and you'll see, the array is always empty :man_shrugging: